ksp_plugin_adapter/reference_frame_selector.cs

```diff
@@ -0,0 +1,134 @@
+﻿#define HAS_SURFACE
+#define HAS_BODY_CENTRED_ALIGNED_WITH_PARENT
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace principia {
+namespace ksp_plugin_adapter {
+
+static class CelestialExtensions {
+  public static bool is_leaf(this CelestialBody celestial) {
+    return celestial.orbitingBodies.Count == 0;
+  }
+
+  public static bool is_root(this CelestialBody celestial) {
+    return celestial.orbit == null;
+  }
+}
+
+class ReferenceFrameSelector {
+
+  private enum FrameType {
+    BODY_CENTRED_NON_ROTATING,
+#if HAS_SURFACE
+    SURFACE,
+#endif
+    BARYCENTRIC_ROTATING,
+#if HAS_BODY_CENTRED_ALIGNED_WITH_PARENT
+    BODY_CENTRED_ALIGNED_WITH_PARENT
+#endif
+  }
+
+  Dictionary<CelestialBody, bool> expanded_;
+  CelestialBody selected_celestial_;
+  FrameType selected_type_ = FrameType.BODY_CENTRED_NON_ROTATING;
+
+  private void RenderSubtree(CelestialBody celestial, int depth) {
+    const int offset = 20;
+    UnityEngine.GUILayout.BeginHorizontal();
+    if (!celestial.is_root()) {
+      UnityEngine.GUILayout.Label(
+          "",
+          UnityEngine.GUILayout.Width(offset * (depth - 1)));
+      if (UnityEngine.GUILayout.Button(
+              celestial.is_leaf() ? "" : (expanded_[celestial] ? "-" : "+"),
```

Comment:
Could we avoid nested ternaries by declaring one or two variables before the if?